typing/types.mli

```diff
@@ -127,16 +127,21 @@ type type_desc =
 
   | Tunivar of string option
   (** Occurrence of a type variable introduced by a
-      forall quantifier / [Tpoly]. *)
+      forall quantifier / [tpoly]. *)
 
-  | Tpoly of type_expr * type_expr list
-  (** [Tpoly (ty,tyl)] ==> ['a1... 'an. ty],
+  | Tpoly of tpoly
+   (** [Tpoly {poly_body : ty; poly_params : tyl}] ==> ['a1... 'an. ty],
       where 'a1 ... 'an are names given to types in tyl
       and occurrences of those types in ty. *)
 
   | Tpackage of Path.t * (string list * type_expr) list
   (** Type of a first-class module (a.k.a package). *)
 
+and tpoly = { poly_body : type_expr; poly_params : type_expr list }
+  (** [{poly_body : ty; poly_params : tyl}] ==> ['a1... 'an. ty],
+      where 'a1 ... 'an are names given to types in tyl
+      and occurrences of those types in ty. *)
```

Comment:
I have doubts about the `poly_params` name: the bound variables are not really parameter? Maybe a name like `poly_vars`, `poly_bound_vars`, or  `poly_univars`  would be a better fit?

Reply:
Actually, here and above, the `:` in the doc seems wrong. Syntactically, either it is `{poly_body; poly_params}` using punning, or `{poly_body = ty; poly_params = tyl}`. The latter does not seem to offer extra information.
I agree that the name should rather be `poly_univars` (they must all be `Tunivar`, or does it change with modular explicits ?).

Reply:
Modular explicits does not impact this at all. Indeed `poly_univars` seems better and I'll do the change while rebasing.